lib/StringClient.js

```diff
@@ -235,6 +260,10 @@ StringClient.prototype.parse = function parse(req, callback) {
             res.on('data', function onData(chunk) {
                 if (hash) {
                     hash.update(chunk);
+
+                    if (hash2) {
```

Comment:
I don't think this needs to be nested in the first if block.

Reply:
Well, I can move it - it just means it'll do more work (always checking the if condition) when not using a content-md5 header. I.e. there should be no hash2 generation when it's not calculating the content-md5.